querydsl-core/src/main/java/com/querydsl/core/group/GroupBy.java

```diff
@@ -106,21 +107,44 @@
     }
 
     /**
-     * Create a new aggregating set expression
+     * Create a new aggregating set expression using a backing LinkedHashMap
      *
      * @param expression
      * @return
      */
     public static <E> AbstractGroupExpression<E, Set<E>> set(Expression<E> expression) {
-        return new GSet<E>(expression);
+        return GSet.createLinked(expression);
     }
 
     public static <E, F> GroupExpression<E, Set<F>> set(GroupExpression<E, F> groupExpression) {
-        return new MixinGroupExpression<E, F, Set<F>>(groupExpression, new GSet<F>(groupExpression));
+        return new MixinGroupExpression<E, F, Set<F>>(groupExpression, GSet.createLinked(groupExpression));
     }
 
     /**
-     * Create a new aggregating map expression
+     * Create a new aggregating set expression using a backing TreeSet
+     *
+     * @param expression
+     * @return
+     */
+    public static <E extends Comparable<? super E>> AbstractGroupExpression<E, Set<E>> sortedSet(Expression<E> expression) {
+        return GSet.createSorted(expression);
+    }
+
+    public static <E, F extends Comparable<? super F>> GroupExpression<E, Set<F>> sortedSet(GroupExpression<E, F> groupExpression) {
+        return new MixinGroupExpression<E, F, Set<F>>(groupExpression, GSet.createSorted(groupExpression));
+    }
+
+    public static <E> AbstractGroupExpression<E, Set<E>> sortedSet(Expression<E> expression, Comparator<? super E> comparator) {
```

Comment:
Does this sound good?
'Create a new aggregating set expression using a backing TreeSet using the given comparator'